05_cleaning_data_11_data_tidying_exercise.Rmd

```diff
@@ -0,0 +1,207 @@
+- What information is in the column `n`?
+- What does `filter(business == "Retail sales, total "| business=="Department stores ")` accomplish?
+
+##### The department store data: `df1`
+
+Having had all the code to wrangle `df2` and being able to understand what each line did, now it's your turn to wrangle a data frame! You'll take  We'll first ask a few questions to get you acquainted with `df1`.
+
+- How many rows are there in `df1`
+- How many columns are there in `df1`
+- In `Jun-1992`, what were retail sales (in millions)?
+
+Now that you have an idea of what data are in the data frame `df`, it's time to wrangle! The goal of wrangling this data set is to mirror (or mimic) what we did in the first data set. This means, that by the end of data wrangling, you should have three columns with the same names as `df_retail`: `business`, `year`, and `n`.
+
+To accomplish this goal, you'll have to figure out what code should replace `function_name` for each line in the code chunk `wrangle-df1`. A different `function_name` will be called in each line of missing code. The arguments necessary to be included within the parentheses will also have to be added.
+
+For example, for the first part of the code, you currently see in the
+
+```r
+df_department <- df1 %>%
+  ## split Period column into one column called "month" and one called "year"
+  function_name()
+```
+
+The comment ` ## split Period column into one column called "month" and one called "year"` tells you what you'll want to accomplish in the next line of code.
+
+So, you may try adding the following:
+
+```r
+df_department <- df1 %>%
+  ## split Period column into one column called "month" and one called "year"
+  separate(Period, into = c('month','year'), extra = 'drop', remove = FALSE)
+```
+
+By changing `function_name()` to the appropriate function `separate()` and including the necessary code within the parentheses, we are able to accomplish exactly what the comment says we wanted to do!
+
+Before you get started on the rest of this,  a quick note on data wrangling : As you master these skills, you *will* write code that errors and things *will* take you multiple tries before you get it right. If you're frustrated, that's ok! It's part of the process. Use the material in this course or places like [StackOverflow](https://stackoverflow.com/) to help you if you find answers if you get stuck!.
+
+OK, now it's your job to work line by line to change `function_name` to the appropriate function in order to accomplish what the comments state you want to accomplish. Work through this code chunk line by line until you have a long data set called `df_department` with three columns:  `business`, `year`, and `n`.
+
+- What function did you use to group the data by the `year` column?
+- What function did you use to add the column name `value`?
+- What function did you use to reorder the column names?
+
+##### Pushing to GitHub
```

Comment:
Should delete Github mentions because we will cover that later. Maybe in the meantime we will have them upload to Google Drive for checking purposes. Need to put some more thought into this.